crates/composefs/src/repository.rs

```diff
@@ -383,15 +401,21 @@ impl<ObjectID: FsVerityHashValue> Repository<ObjectID> {
         self.write_image(Some(name), &data)
     }
 
-    pub fn open_image(&self, name: &str) -> Result<OwnedFd> {
+    fn open_image(&self, name: &str) -> Result<(OwnedFd, bool)> {
         let image = self.openat(&format!("images/{name}"), OFlags::RDONLY)?;
 
-        if !name.contains("/") {
+        if !name.contains("/") && !self.insecure {
             // A name with no slashes in it is taken to be a sha256 fs-verity digest
             ensure_verity_equal(&image, &ObjectID::from_hex(name)?)?;
         }
 
-        Ok(image)
+        match measure_verity::<ObjectID>(&image) {
```

Comment:
You're effectively measuring and comparing verity twice here... You left the check above...